src/cmd-compress

```diff
@@ -107,20 +104,16 @@ def compress_one_builddir(builddir):
     if len(args.artifact) > 0:
         only_artifacts = set(args.artifact)
 
-    for img_format in buildmeta['images']:
-        if img_format in imgs_to_skip:
+    for img_format, img in buildmeta['images'].items():
+        if img.get('skip-compression', False):
             print(f"Skipping {img_format}")
             continue
         if only_artifacts is not None and img_format not in only_artifacts:
             continue
 
-        img = buildmeta['images'][img_format]
         file = img['path']
         filepath = os.path.join(builddir, file)
-        # check if it's not already compressed, even if with a different
-        # compressor (like for GCP, which needs .tar.gz, see:
-        # https://github.com/coreos/coreos-assembler/pull/1009)
-        if not file.endswith(tuple(ext_dict.values())):
+        if img.get('uncompressed-sha256') is None:
```

Comment:
note.. IIUC this requires us to make sure we remove `uncompressed-sha256` when we `cosa decompress` in the future, which is OK.

Though, I almost look at `uncompressed-sha256` as something that shouldn't change and could be considered immutable. Could we instead do something like:

```
 if img.get('uncompressed-sha256') is None or
 img.get('uncompressed-sha256') == img.get('sha256'):
```

Reply:
It's a fair question.  Currently we only set `uncompressed-sha256` on artifacts that are compressed.  I think there's value to retaining that invariant, but another possible approach is to set it everywhere.